### ISSUE TITLE
Seeding parameter values and prunning trials

### ISSUE THREAD

**Issue by amitchandak** [facebook/Ax#378]
Hi,
How can I seed the parameter values before calling the "optimize" function? 
For example in  [tune_cnn_example](https://github.com/facebook/Ax/blob/master/tutorials/tune_cnn.ipynb) , if we know that optimal learning rate will be around say 0.5, then can we seed it with value and define range like [0.3, 0.8] ? 

Also is there a way to prune trials which we know might not give optimal  hyper-parameter values ? 

Thank You

**Comment by 2timesjay** [comment]
In https://github.com/facebook/Ax/blob/master/tutorials/tune_cnn.ipynb you can see the search space set up with "bounds" specified. In cell [7]  you see

` {"name": "lr", "type": "range", "bounds": [1e-6, 0.4], "log_scale": True},`

You could change the range to [0.3, 0.8] by setting

`{"name": "lr", "type": "range", "bounds": [0.3, 0.8], "log_scale": True},`

Trials outside this bound will never be generated.

**Comment by amitchandak** [comment]
@2timesjay thank you for answering. I have been doing what your have suggested but in suggested approach the BO algorithm will start with random values chosen from the set. It will be better to start with "lr=0.5" and then explore around this value but constraint by the defined range. 

Also can you please suggest how can I define "pruning" strategies in BoTorch or Ax framework ? 
Something like this : https://github.com/optuna/optuna/blob/master/tutorial/007_pruning.py 

Thanks for your time,

**Comment by 2timesjay** [comment]
@amitchandak thank you for the follow-up. 

To seed with what you know to be good values, you could use the service API (tutorial: https://ax.dev/tutorials/gpei_hartmann_service.html) and manually add an initial set of trials based on good parameters. We don't otherwise have any feature for  "hinting" or priors because we rely on BO after initialization.

Unfortunately we don't support anything exactly like Optuna's pruning right now; you could add monitoring with some rule for early stopping yourself by using the service or dev APIs and adding that code to their trial generation loops.

Any more details on your use case - the cost and duration of each trial for instance and the application domain?

**Comment by amitchandak** [comment]
@2timesjay thanks for your reply. 
Application domain is to figure out right hyper-parameters like learning-rate, momentum value and weight-decay for training different models (VGG16, Pres-Net and Wide-ResNet) on CIFAR-10 and CIFAR-100. 

I have figured values of these parameters through grid-search and hence wanted to seed the BO routine with those values to start of. 

Also since few models like Wide-ResNet takes about 21 hrs for 100 Trials on my GPU, i wanted to see if pruning can help to speed up this. 

Thank you for your inputs, I will try what you have suggested. If you don't have any other suggestions then I will close this ticket.

**Comment by 2timesjay** [comment]
@amitchandak then I can confirm that following the service API tutorial above and using 'attach_trial' and 'complete_trial' with your previous grid search is the best option.

with 100 trials you'll have a very strong initial fit for bayesopt.

Avoid intializing with additional unnecessary random trials by passing an extra kwarg:

`ax_client.create_experiment(..., choose_generation_strategy_kwargs={"num_initialization_trials": 1})`

(0 won't work because of validation, unfortunately, but see https://github.com/facebook/Ax/blob/master/ax/modelbridge/dispatch_utils.py and https://github.com/facebook/Ax/blob/master/ax/modelbridge/generation_strategy.py for the code that manages these settings)
